sql/moz-fx-data-shared-prod/telemetry_derived/firefox_android_clients_v1/query.sql

```diff
@@ -0,0 +1,482 @@
+        first_session.adjust_network,
+        first_session.first_run_date,
+        metrics.adjust_network,
+        metrics.metrics_ping_submission_date
+      ) AS adjust_network__source_ping_date,
+      get_first_not_null_date(
+        existing.install_source,
+        existing.install_source__source_ping_date,
+        NULL,
+        NULL,
+        metrics.install_source,
+        metrics.metrics_ping_submission_date
+      ) AS install_source__source_ping_date
+    ) AS metadata
+  FROM
+    existing
+  LEFT JOIN
+    first_session_ping first_session
+  USING
+    (client_id)
+  LEFT JOIN
+    metrics_ping_unique AS metrics
+  USING
+    (client_id)
+),
+first_seen_with_pings_new_data AS (
+  SELECT
+    first_seen.client_id,
+    first_seen.first_seen_date,
+    first_session.first_run_date AS first_run_date,
+    first_session.first_reported_country AS first_reported_country,
+    first_session.first_reported_isp AS first_reported_isp,
+    first_session.channel AS channel,
+    first_session.device_manufacturer AS device_manufacturer,
+    first_session.device_model AS device_model,
+    first_session.os_version AS os_version,
+    first_session.adjust_campaign AS adjust_campaign,
+    first_session.adjust_ad_group AS adjust_ad_group,
+    first_session.adjust_creative AS adjust_creative,
+    COALESCE(first_session.adjust_network, metrics.adjust_network) AS adjust_network,
+    metrics.install_source AS install_source,
+    STRUCT(
+      CASE
+      WHEN
+        first_session.client_id IS NULL
+      THEN
+        FALSE
+      ELSE
+        TRUE
+      END
+      AS reported_first_session_ping,
+      CASE
+      WHEN
+        metrics.client_id IS NULL
+      THEN
+        FALSE
+      ELSE
+        TRUE
+      END
+      AS reported_metrics_ping,
+      first_session.first_run_date AS min_first_session_ping_date,
+      CASE
+      WHEN
+        first_session.adjust_network IS NOT NULL
+      THEN
+        'first_session'
+      WHEN
+        first_session.adjust_network IS NULL
+        AND metrics.adjust_network IS NOT NULL
+      THEN
+        'metrics'
+      ELSE
+        NULL
+      END
+      AS adjust_network__source_ping,
+      'metrics' AS install_source__source_ping,
+      CASE
+      WHEN
+        first_session.adjust_network IS NOT NULL
+      THEN
+        first_session.first_run_date
+      WHEN
+        first_session.adjust_network IS NULL
+        AND metrics.adjust_network IS NOT NULL
+      THEN
+        metrics.metrics_ping_submission_date
+      ELSE
+        NULL
+      END
+      AS adjust_network__source_ping_date,
+      CASE
+      WHEN
+        metrics.install_source IS NOT NULL
+      THEN
+        metrics.metrics_ping_submission_date
+      ELSE
+        NULL
+      END
+      AS install_source__source_ping_date
+    ) AS metadata
+  FROM
+    first_seen
+  LEFT JOIN
+    first_session_ping first_session
+  USING
+    (client_id)
+  LEFT JOIN
+    metrics_ping_unique AS metrics
```

Comment:
Can we verify the synchrony of client_ids in metrics -> baseline, and first_session -> baseline? Just to see how many we'll leave out with this left join